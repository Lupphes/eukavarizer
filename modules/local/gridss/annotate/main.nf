process GRIDSS_ANNOTATE {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container 'quay.io/biocontainers/bioconductor-structuralvariantannotation:1.22.0--r44hdfd78af_0'

    input:
    tuple val(meta), path(vcf)

    output:
    tuple val(meta), path("*.reclassified.vcf"), emit: vcf
    path "*.simple.bed"                        , emit: bed
    path "versions.yml"                        , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    gunzip -c ${vcf} > input.vcf

    # Check if VCF has any variants (non-header lines)
    # Use || true to prevent grep from causing pipeline failure when no matches found
    VARIANT_COUNT=\$(grep -v "^#" input.vcf | wc -l || true)

    if [ "\${VARIANT_COUNT}" -eq 0 ]; then
        # No variants - create empty output files with proper headers
        grep "^#" input.vcf > ${prefix}.reclassified.vcf
        touch ${prefix}.simple.bed
        echo "No variants found in VCF. Created empty output files." >&2
    else
        # Process normally with R script
        simple-event-annotation.R ${args} input.vcf ${prefix}.reclassified.vcf ${prefix}.simple.bed
    fi

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$( R --version 2>&1 | head -n1 | sed 's/R version //; s/ .*//' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.reclassified.vcf
    touch ${prefix}.simple.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        r-base: \$( R --version 2>&1 | head -n1 | sed 's/R version //; s/ .*//' )
    END_VERSIONS
    """
}
