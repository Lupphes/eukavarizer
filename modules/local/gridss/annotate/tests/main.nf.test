nextflow_process {

    name "Test Process GRIDSS_ANNOTATE"
    script "../main.nf"
    process "GRIDSS_ANNOTATE"
    tag "gridss"
    tag "gridss/annotate"
    tag "modules"
    tag "modules_local"

    test("Annotate GRIDSS VCF - success") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'test' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/test_annotate.vcf.gz')
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert process.out.vcf },
                { assert process.out.vcf.get(0).get(1).toString().endsWith('.vcf') },
                { assert process.out.bed },
                { assert process.out.bed.get(0).toString().endsWith('.bed') },
                { assert process.out.versions }
            )
        }
    }

    test("Annotate empty VCF - should handle gracefully") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of([
                    [ id: 'empty' ],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/vcf/empty.vcf.gz')
                ])
                """
            }
        }

        then {
            assertAll(
                { assert process.success || process.failed },
                { assert snapshot(process.out).match() },
                { assert process.out.versions }
            )
        }
    }
}
