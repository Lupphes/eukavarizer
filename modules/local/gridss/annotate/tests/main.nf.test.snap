{
    "Annotate GRIDSS VCF - success": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.reclassified.vcf:md5,78d481b1ed492ccecd55e04eaeacfcad"
                    ]
                ],
                "1": [
                    "test.simple.bed:md5,23f7365dad53175bd052c2ea4a29106c"
                ],
                "2": [
                    "versions.yml:md5,4116918df4efe8fb0febdefe97902d35"
                ],
                "bed": [
                    "test.simple.bed:md5,23f7365dad53175bd052c2ea4a29106c"
                ],
                "vcf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.reclassified.vcf:md5,78d481b1ed492ccecd55e04eaeacfcad"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4116918df4efe8fb0febdefe97902d35"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T22:35:45.501180473"
    },
    "Annotate empty VCF - should handle gracefully": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "empty"
                        },
                        "empty.reclassified.vcf:md5,596e04275526577e5bf3724915160281"
                    ]
                ],
                "1": [
                    "empty.simple.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,4116918df4efe8fb0febdefe97902d35"
                ],
                "bed": [
                    "empty.simple.bed:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "vcf": [
                    [
                        {
                            "id": "empty"
                        },
                        "empty.reclassified.vcf:md5,596e04275526577e5bf3724915160281"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,4116918df4efe8fb0febdefe97902d35"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.8"
        },
        "timestamp": "2025-10-20T22:35:52.020024741"
    }
}